File: code/backend/src/main/java/com/blog/entity/Music.java
package com.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "music")
public class Music {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 200)
    private String title;
    
    @Column(nullable = false, length = 100)
    private String artist;
    
    @Column(length = 200)
    private String album;
    
    @Column(nullable = false)
    private Integer duration;
    
    @Column(nullable = false, length = 500)
    private String url;
    
    @Column(name = "cover_url", length = 500)
    private String coverUrl;
    
    @Column(name = "display_order", nullable = false)
    private Integer displayOrder = 1;
    
    @Column(name = "uploaded_at")
    private LocalDateTime uploadedAt;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: code/frontend/src/api/music.ts
import request from '@/utils/request'
import { IApiResponse } from '@/types'

export interface Music {
  id: number
  title: string
  artist: string
  album: string | null
  duration: number
  url: string
  coverUrl: string | null
  displayOrder: number
  uploadedAt: string | null
  createdAt: string
  updatedAt: string
}

export interface MusicCreateRequest {
  title: string
  artist: string
  album?: string
  duration: number
  url: string
  coverUrl?: string
  displayOrder?: number
}

export interface MusicUpdateOrderRequest {
  displayOrder: number
}

export const musicApi = {
  getAll: () => request.get<Music[]>('/api/music'),
  
  getById: (id: number) => request.get<Music>(`/api/music/${id}`),
  
  create: (data: MusicCreateRequest) => request.post<Music>('/api/music', data),
  
  update: (id: number, data: MusicCreateRequest) => request.put<Music>(`/api/music/${id}`, data),
  
  updateOrder: (id: number, data: MusicUpdateOrderRequest) => request.put<Music>(`/api/music/${id}/order`, data),
  
  delete: (id: number) => request.delete<void>(`/api/music/${id}`),
}
